lib/util/git/index.ts

```diff
@@ -366,7 +372,7 @@ export async function cloneSubmodules(
     try {
       logger.debug(`Cloning git submodule at ${submodule}`);
       await gitRetry(() =>
-        git.submoduleUpdate(['--init', '--recursive', submodule]),
+        git.env(gitEnv).submoduleUpdate(['--init', '--recursive', submodule]),
```

Comment:
```suggestion
        git.env(getGitEnvironmentVariables()).submoduleUpdate(['--init', '--recursive', submodule]),
```

is sufficient to get cloning of private submodules to work. Passing process.env doesn't seem necessary